src/Microsoft.ComponentDetection.Detectors/dockerfile/DockerfileComponentDetector.cs

```diff
@@ -69,25 +69,33 @@ private Task ParseDockerFile(string fileContents, string fileLocation, ISingleFi
 
         private DockerReference ProcessDockerfileConstruct(DockerfileConstruct construct, char escapeChar, Dictionary<string, string> stageNameMap)
         {
-            var instructionKeyword = construct.Type;
-            DockerReference baseImage = null;
-            if (instructionKeyword == ConstructType.Instruction)
-            {   
-                var constructType = construct.GetType().Name;
-                switch (constructType)
+            try
+            {
+                var instructionKeyword = construct.Type;
+                DockerReference baseImage = null;
+                if (instructionKeyword == ConstructType.Instruction)
                 {
-                    case "FromInstruction":
-                        baseImage = ParseFromInstruction(construct, escapeChar, stageNameMap);
-                        break;
-                    case "CopyInstruction":
-                        baseImage = ParseCopyInstruction(construct, escapeChar, stageNameMap);
-                        break;
-                    default:
-                        break;
+                    var constructType = construct.GetType().Name;
+                    switch (constructType)
+                    {
+                        case "FromInstruction":
+                            baseImage = ParseFromInstruction(construct, escapeChar, stageNameMap);
+                            break;
+                        case "CopyInstruction":
+                            baseImage = ParseCopyInstruction(construct, escapeChar, stageNameMap);
+                            break;
+                        default:
+                            break;
+                    }
                 }
-            }
 
-            return baseImage;
+                return baseImage;
+            } catch (Exception e)
```

Comment:
Add a test for your changes

Reply:
@FernandoRojo  is going to fast follow with unit tests